### ISSUE TITLE
Merge-to-mainline, build, and release process needs to be defined

### ISSUE THREAD

**Issue by mjwesterhof** [UniversalDevicesInc/Polyglot#27]
We need a documented process and/or checklist and/or Makefile rules to help perform all the necessary actions required to merge changes into the main branch, build on the Pi, Mac, and PC platforms, commit the results, and tag that version on the mainline.

Let the discussion commence!

**Comment by Einstein42** [comment]
I say we create an 'unstable-rc' branch. Once we have added a feature or fixed something we merge to unstable-rc... all the developers should be running unstable-rc... we put a change freeze on on unstable-rc after the merge is performed for 2 days. If no issues arrise we can build and push to unstable-release and reopen unstable-rc for new features/fixes.

**Comment by mkohanim** [comment]
Hi James,
That’s a great idea!

With kind regards,

---

  Michel Kohanim
  CEO

  (p) 818.631.0333
  (f)  818.436.0702
  http://www.universal-devices.comhttp://www.universal-devices.com/

**Comment by Einstein42** [comment]
https://github.com/UniversalDevicesInc/Polyglot/wiki/Build-and-Release-Information

Anything else you guys want on that?

**Comment by mkohanim** [comment]
Hi James,

Thank you. This is excellent. I am sending an email blast. With regards to the documentation, I shall have it pointed to:
https://github.com/UniversalDevicesInc/Polyglot/blob/unstable-release/README.md … is that correct?

With kind regards,

---

  Michel Kohanim
  CEO

  (p) 818.631.0333
  (f)  818.436.0702
  http://www.universal-devices.comhttp://www.universal-devices.com/

**Comment by Einstein42** [comment]
Yes sir.

**Comment by mkohanim** [comment]
Thank you.

With kind regards,

---

  Michel Kohanim
  CEO

  (p) 818.631.0333
  (f)  818.436.0702
  http://www.universal-devices.comhttp://www.universal-devices.com/

**Comment by mjwesterhof** [comment]
(Apologize if this is already documented; I'm at work and can't review in depth until this evening) -- we should make sure we increment the version number in version.py at the correct times, so that we know what the user is running from the logs... otherwise this could get really confusing fast!

Last I checked, the development head was at 0.0.1 - the version number is a string so we can easily put "rc", or "beta" suffixes on if desired.  It's also in a file all by itself so the inevitable manual merges we'll get when merging git branches will end up isolated and easy to resolve.

**Comment by Einstein42** [comment]
Let me understand mjwesterhof. You modified everything to use the PGVERSION imported from version.py so we only need to update it there correct? 

**Edit** I checked he did. That's great. Thanks for doing that. We will increment a single number in the third octet per merge to unstable-rc. That work for everyone?

**Comment by mjwesterhof** [comment]
Works for me!

**Comment by Einstein42** [comment]
Ok. This is now in effect! Go team.

**Comment by mjwesterhof** [comment]
@mkohanim - Can you create a repository alongside this one for the binaries and documentation?  This would allow us to remove the generated artifacts from the source repo, and then we'd simply check them in and tag them in the binary repo.

Users would never need to check out or deal with the source repo; they'd only ever reference the release repo for the docs and pyz files.

Thoughts and comments from the team welcomed!

**Comment by mkohanim** [comment]
@mjweterhof, of course. Do you mean something like UniversalDevicesInc/Polyglot-Bin? 

With kind regards,
Michel

**Comment by Einstein42** [comment]
I think that is fine, or even 'Polyglot-Release'

**Comment by mkohanim** [comment]
Done! Thank you.

**Comment by jimboca** [comment]
@mjwesterhof Can we merge the current development branch up the chain?  I would like to release my camera server and don't want to ask everyone to grab the development branch?

**Comment by Einstein42** [comment]
@jimboca go ahead and pull dev into unstable-rc please

**Comment by jimboca** [comment]
done
